### DISCUSSION TITLE
Why can't wat2wasm compile the add_one() example in the README?

### DISCUSSION THREAD

**Discussion by drauschenbach** [wasmerio/wasmer#4421] [General]
Just trying to understand the formats and file extensions. On a Mac:

```sh
$ vi add_one.wat
(module
(type $t0 (func (param i32) (result i32)))
(func $add_one (export "add_one") (type $t0) (param $p0 i32) (result i32)
    get_local $p0
    i32.const 1
    i32.add))

$ brew install wabt
$ wat2wasm add_one.wat -o add_one.wasm
add_one.wat:4:5: error: unexpected token get_local, expected ).
    get_local $p0
    ^^^^^^^^^
```

**Comment by drauschenbach** [comment]
FYI:

> Please see https://github.com/WebAssembly/wabt/issues/1883 -- get_local isn't a valid instruction in WebAssembly. (It used to be called that when WebAssembly was in draft form, but the names were changed in 2018 before the WebAssembly 1.0 spec was published, and wabt dropped support for the draft names in 2021 (https://github.com/WebAssembly/wabt/pull/1792).)

https://github.com/WebAssembly/wabt/issues/1802#issuecomment-1236657208

**Reply by theduke** [reply]
Yeah, I think this is just a really old example.
